xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml

```diff
@@ -645,6 +646,21 @@ $services.localization.render('core.users.unknownUser')##
     #set ($discard = $query.bindValue('qf', $boost))
   #end
 #end
+    
+#macro (setPhraseFields $query)
+  #if ($solrConfig.phraseFields.substring(0, 0) == '')
+    ## If the value of the 'queryFields' parameter is a string then it means that the same query fields are used for
+    ## all result types.
+    #set ($phraseFieldsBoost = $solrConfig.phraseFields)
+  #else
+    ## There are different query fields for each result type.
+    #set ($phraseFieldsBoost = $solrConfig.phraseFields.get($type))
+  #end
+  #if ("$!phraseFieldsBoost" != '')
+    #set ($discard = $query.bindValue('pf', $phraseFieldsBoost))
+    #set ($discard = $query.bindValue('ps', '10'))
```

Comment:
We could make the phrase slop configurable also.

Reply:
True. I'll implement.

Reply:
done